ggshield/git_shell.py

```diff
@@ -1,11 +1,26 @@
+import os
 import subprocess
+from functools import lru_cache
+from shutil import which
 from typing import Any, List
 
 import click
 
 
 COMMAND_TIMEOUT = 45
-GIT_PATH = "git"
+
+
+@lru_cache(None)
+def get_git_path(cwd: str) -> str:
+    git_path = str(which("git"))
+
+    if cwd in git_path:
```

Comment:
`if cwd in git_path && cwd not in env PATH variable` would solve edge case as well when cwd is also in PATH explicitly

Reply:
Added that small edge case in the latest commit, thanks